packages/amplify-graphql-api-construct/src/internal/metadata.ts

```diff
@@ -0,0 +1,36 @@
+import { AuthorizationModes, IAmplifyGraphqlDefinition } from '../types';
+
+export const getMetadataDataSources = (definition: IAmplifyGraphqlDefinition): string => {
+  const dataSourceDbTypes = Object.values(definition.dataSourceStrategies).map((strategy) => strategy.dbType.toLocaleLowerCase());
+  const customSqlDbTypes = (definition.customSqlDataSourceStrategies ?? []).map((strategy) => strategy.strategy.dbType.toLocaleLowerCase());
+  const dataSources = [...new Set([...dataSourceDbTypes, ...customSqlDbTypes])].sort();
+  return dataSources.join(',');
+};
+
+export const getMetadataAuthorizationModes = (authorizationModes: AuthorizationModes): string => {
+  const configKeyToAuthMode: Record<string, string> = {
+    iamConfig: 'aws_iam',
+    oidcConfig: 'openid_connect',
+    identityPoolConfig: 'amazon_cognito_identity_pools',
+    userPoolConfig: 'amazon_cognito_user_pools',
+    apiKeyConfig: 'api_key',
+    lambdaConfig: 'aws_lambda',
+  };
+  const authModes = Object.keys(authorizationModes)
```

Comment:
One of the tests above asserts the order of the auth mode metric. Is that intentional? Should we sort the values here so we can provide a stable value irrespective of the order in which they are declared?

Reply:
Good idea, I will apply a sort.